aya_tamimi/task_manager.py

```diff
@@ -0,0 +1,203 @@
+        try:
+            utils.get_valid_input(description)
+            break
+        except ValueError as e:
+            print(e)
+
+    while True:
+        due_date = input("Enter task due_date (in YYYY-MM-DD format): ")
+        try:
+            utils.validate_date(due_date)
+            break
+        except ValueError as e:
+            print(e)
+    while True:
+        status = input("Enter task status (InProgress , Completed): ").strip().lower()
+        try:
+            utils.validate_status(status)
+            break
+        except ValueError as e:
+            print(e)
+
+    is_urgent = input("is the task urgent(yes/no) ? ").strip().lower()
+    if is_urgent in ['yes' ,'y']:
+        while True:
+            priority = input("Enter task priority (High/Medium/Low): ").strip().lower()
+            try:
+                utils.validate_priority(priority)
+                break
+            except ValueError as e:
+                print(e)
+        task_instance = UrgentTask(priority ,title,description,due_date,status)
+        tasks.append(task_instance)
+        print("Urgent task successfully created.")
+        main()
+    else :
+        task_instance = Task(title,description,due_date,status)
+        tasks.append(task_instance)
+        print("task successfully created")
+        main()
+
+def view_tasks():
+    if not tasks:
+        print("🚫 No tasks available. 🚫")
+    else:
+        urgent_tasks = [task for task in tasks if isinstance(task, UrgentTask)]
+        regular_tasks = [task for task in tasks if isinstance(task, Task) and not isinstance(task, UrgentTask)]
+
+        if urgent_tasks:
+            print("\n" + "="*30)
+            print("🚨🚨🚨 Urgent Tasks 🚨🚨🚨")
+            print("="*30)
+            for task in urgent_tasks:
+                task.display()
+                print("\n")
+
+            print("\n")
+        if regular_tasks:
+            print("\n" + "-"*30)
+            print("📝📝📝 Regular Tasks 📝📝📝")
+            print("-"*30)
+            for task in regular_tasks:
+                task.display()
+                print("\n")
+    main()
+
+
+# This function find a specific task in tasks list and return it to make operation on it 
+def find_task_by_id(id):
+    for task in tasks :
+        if task.task_id == id:
+            return task
+    raise ValueError("Task with the provided ID does not exist.")
+
+# This function update details for any task
+def update_task():
+    task_id = input("Please Enter the Task ID you want to update: ").strip()
+    try:
+        task_to_update = find_task_by_id(task_id)        
+        print("\nPlease provide the new details. Leave blank if you don't want to change a specific field.") 
+        new_title = input("New Title (leave blank to keep the current title): ").strip()
+        new_status = input("New Status (leave blank to keep the current status): ").strip()
+        if new_status:
+            utils.validate_status(new_status)    
```

Comment:
Usually, in the backend, we receive values first and then validate them.

I understand that you're validating the values directly here because the user is entering them, but I prefer the approach used in adding tasks.

In the add task method, if the user enters an incorrect value, they have a chance to re-enter it. However, in this case, if the value is incorrect, the program exits.

Reply:
fixed it , Now users can re-enter values if they mess up